bacon/model/StorageManager.swift

```diff
@@ -2,14 +2,27 @@
 //  StorageManager.swift
 //  bacon
 //
+//  An API for all storage related functionalities.
+//  Provides an abstraction over the underlying storage library dependacies.
+//
 //  Created by Travis Ching Jia Yea on 19/3/19.
 //  Copyright © 2019 nus.CS3217. All rights reserved.
 //
 
 import Foundation
 
-protocol StorageManager {
-    // API to be called embedded MongoDB
-    func saveTransaction()
-    // func loadTransaction() with different filters
+class StorageManager {
```

Comment:
Instead of having a proxy class to delegate methods to concreteStorage, why not make this a protocol, and have concreteStorage conform to this protocol? That way, we eliminate the proxy, but we also work with interfaces.

Reply:
When creating a storage inside other modules, for instance in ModelManager say you wan to create a Database instance you will then have to do, let database: StorageManager = ConcreteDatabase().
But what if, later on we want to change database library, to ConcreteDatabaseSQL(). Instead of having to delete all the code in the current ConcreteDatabase() I would prefer to create a new swift file. Hence having StorageManager as a class and not a protocol, makes any changes to the database local to this class. ModalManager would not need to change its database variable to instantiate ConcreteDatabaseSQL().